testing/test_pdb.py

```diff
@@ -518,6 +518,43 @@ def test_1():
         assert "1 failed" in rest
         self.flush(child)
 
+    def test_pdb_interaction_continue_recursive(self, testdir):
+        p1 = testdir.makepyfile(
+            """
+            import pytest
```

Comment:
Strange but I get (apparently) the same interactive session you describe here in this tests *without* your patch:

```
======================== test session starts ========================
platform win32 -- Python 3.6.6, pytest-3.10.1.dev27+gf06fe436.d20181108, py-1.6.0, pluggy-0.8.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('c:\\pytest\\.hypothesis\\examples')
rootdir: c:\pytest\.tmp, inifile: pytest.ini
plugins: xdist-1.23.2, parallel-0.0.6, forked-0.2, hypothesis-3.74.2
collected 1 item

.tmp\test-recursive-pdb.py
>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>
> c:\pytest\.tmp\test-recursive-pdb.py(12)test_1()
-> x = 3
(Pdb) debug foo()
ENTERING RECURSIVE DEBUGGER
> <string>(1)<module>()
((Pdb)) c
print_from_foo
LEAVING RECURSIVE DEBUGGER
(Pdb) c

>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>
F                                   [100%]

============================= FAILURES ==============================
______________________________ test_1 _______________________________

    def test_1():
        i = 0
        print("hello17")
        pytest.set_trace()
>       x = 3
E       assert 0

.tmp\test-recursive-pdb.py:12: AssertionError
----------------------- Captured stdout call ------------------------
hello17
hello18
===================== 1 failed in 61.35 seconds =====================
``` 

Am I missing something? 🤔

Reply:
Thanks for testing.
That looks like it is expected, but did not work for me.
Maybe also due to pdbpp instead of pdb.

OTOH, the inner "c" without this patch should get handled with the "do_continue" really though (and then resume capturing, causing pdb to fail then).

Reply:
At least the new test is failing without the patch:

```
        child.expect("LEAVING RECURSIVE DEBUGGER")
>       assert b"PDB continue" not in child.before
E       AssertionError: assert b'PDB continue' not in b'++)) c\r\n\r\n>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>\r\n\x1b[...--------------------------- Captured stdout call ------------------------------------\r\nhello17\r\nprint_from_foo\r\n'
E        +  where b'++)) c\r\n\r\n>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>\r\n\x1b[...--------------------------- Captured stdout call ------------------------------------\r\nhello17\r\nprint_from_foo\r\n' = <pexpect.pty_spawn.spawn object at 0x7fd6ecdf8400>.before
```

Reply:
Maybe capturing is stacked in some way on Windows already?

Reply:
Additionally I've thought about having some new attribute to see the level/stacking of suspend/resume on the capman, but it would not really help here - more relevant for #4331 probably.

Reply:
Hmm, pexpect does not run on AppVeyor?!
(since ec6fca4a)